microprofile/health-smallrye/src/main/java/org/wildfly/extension/microprofile/health/MicroProfileHealthReporter.java

```diff
@@ -38,8 +39,21 @@ public class MicroProfileHealthReporter {
     private final HealthCheck emptyDeploymentLivenessCheck;
     private final HealthCheck emptyDeploymentReadinessCheck;
     private final HealthCheck emptyDeploymentStartupCheck;
-
     private boolean userChecksProcessed = false;
+    private final AtomicReference<String> defaultProceduresDeactivatorReference = new AtomicReference<>();
+
+    public void deactivateDefaultProcedures(final String deploymentName) {
+        // If a deployment disables default procedures, that will affect all the default procedures,
+        //  including those relating to the global server context, see https://issues.redhat.com/browse/WFLY-19147
+        if (this.defaultProceduresDeactivatorReference.get() == null) {
+            MicroProfileHealthLogger.LOGGER.deploymentDisablingDefaultProcedures(deploymentName);
+        }
+        this.defaultProceduresDeactivatorReference.compareAndSet(null, deploymentName);
```

Comment:
I'm not sure why you need to keep the track of the deployment name?

Reply:
I though it could be useful anyway, e.g. improving subsequent log messages that would read such name.